### PR TITLE
Split up action definitions into template and pre-configured image in golangci

### PR DESCRIPTION
**Pull request by sagikazarmark**
Trying to apply the patterns discussed in #2781

cc @gerhard @helderco

### CODE REVIEW

File: pkg/universe.dagger.io/alpha/go/golangci/lint.cue [hunk 1]
```diff
@@ -8,8 +8,14 @@ import (
...
 
+// Like #LintBase, but with a pre-configured container image.
+#Lint: #LintBase & {
+	_image: #Image
+	input:  _image.output
+}
+
 // Lint using golangci-lint
-#Lint: {
+#LintBase: {
```
Comment (sagikazarmark) [inline pkg/universe.dagger.io/alpha/go/golangci/lint.cue:18]: Some alternatives I came up with in the discussion:

- Template (eg. `LintTemplate`)
- Command (eg. `LintCommand`)
- Definition (or Def) (eg. `LintDefinition`) (this is somewhat redundant)
- Proto (eg. `LintProto`)

For some reason, Template and Definition are the closest to me, but I can't reason about them. WDYT?

Comment (grouville) [reply pkg/universe.dagger.io/alpha/go/golangci/lint.cue:17]: I personally like it

**Review by grouville — APPROVED** [review]
LGTM. Please note that I will first merge #2210, which will impact this PR (slightly). You'll have to rebase

**Comment by grouville** [comment]
@sagikazarmark you can rebase

**Comment by sagikazarmark** [comment]
Thanks @grouville . Rebase done.

**Review by grouville — COMMENTED** [review]
LGTM

**Comment by sagikazarmark** [comment]
Thanks for merging @grouville !

I actually wanted this to serve as an example for the discussion we had earlier. If there is a consensus that this pattern will work for now, I'll start making a few PRs adding more tools/changing existing ones. Maybe also add some documentation.

Is that the case? Or do you want to explore more options?

cc @gerhard @helderco

**Comment by gerhard** [comment]
I ran out of time this week, back in ~1 week.

**Pull request merged**